src/app/core/services/contract.service.ts

```diff
@@ -77,11 +68,33 @@ export class ContractService extends CommonService {
       name = '%' + keyword + '%';
       updateQuery = `_and: {_or: [{name: {_like: "${name}"}}, {code_id: {_eq: ${codeId}}}]},`;
     } else if (keyword?.length > 0) {
-      name = '%' + keyword + '%';
-      updateQuery = `name: {_ilike: "${name}"},`;
+      filterName = `_ilike: "${'%' + keyword + '%'}"`;
     } else {
       updateQuery = '';
     }
+
+    let typeQuery = 'code: {_or: [{type: {_in: $type}}, {_and: {type: {_is_null: true}}}]}';
+    if (isFilterCW4973) {
+      if (contractType?.length >= 2) {
+        filterName = filterName?.length > 0 ? filterName : '_eq: "crates.io:cw4973"';
+        typeQuery = contractType?.includes('')
+          ? `_or: [{code: {_or: [{type: {_in: $type}}, {_and: {type: {_is_null: true}}}]}}, {name: {${filterName}}}],`
+          : `_or: [{code: {type: {_in: $type}}}, {name: {${filterName}}}],`;
+      } else {
+        filterName = ', ' + filterName;
+        typeQuery = contractType?.includes('')
```

Comment:
Sử dụng constant hoặc đặt biến constant để dễ đọc code
```suggestion
        const FILTER_ALL = ''; 
        typeQuery = contractType?.includes(FILTER_ALL)
```